datacube/drivers/postgres/_api.py

```diff
@@ -535,6 +536,77 @@ def search_datasets(self, expressions,
                                                   select_fields, with_source_ids, limit)
         return self._connection.execute(select_query)
 
+    @staticmethod
+    def search_unique_datasets_query(expressions, select_fields, limit):
+        """
+        We are not dealing with dataset_source table here and we are not joining
+        dataset table with dataset_location table. We are aggregating stuff
+        in dataset_location per dataset basis if required. It returns the construted
+        query.
+        """
+
+        # expressions involving DATASET_SOURCE cannot not done for now
+        for expression in expressions:
+            assert expression.field.required_alchemy_table != DATASET_SOURCE,\
+                'Joins with dataset_source cannot be done for this query'
+
+        # expressions involving 'uri' and 'uris' will be handled different
+        expressions = [expression for expression in expressions
+                       if expression.field.required_alchemy_table != DATASET_LOCATION]
+
+        if select_fields:
+            select_columns = []
+            for field in select_fields:
+                if field.name in {'uri', 'uris'}:
+                    # All active URIs, from newest to oldest
```

Comment:
this part is not being tested, please add tests for uri fields

Reply:
Done